pkg/registry/bundlegraphloader_test.go

```diff
@@ -312,3 +313,73 @@ func TestBundleGraphLoader(t *testing.T) {
 		})
 	}
 }
+
+func TestIsSkipPatchCandidate(t *testing.T) {
+	tests := []struct{
+		name     string
+		added     string
+		compare  string
+		expected bool
+		commutative bool
+	}{
+		{
+			name: "equal versions",
+			added: "0.0.0",
+			compare: "0.0.0",
+			expected: false,
+			commutative: true,
+		},
+		{
+			name: "do not accept different major/minor version",
+			added: "0.1.0",
+			compare: "0.2.0",
+			expected: false,
+			commutative: true,
+		},
+		{
+			name: "accept larger patch version",
+			added: "0.0.1",
+			compare: "0.0.0",
+			expected: true,
+		},
+		{
+			name: "accept patch version without pre-release",
+			added: "0.0.0",
+			compare: "0.0.0-1",
+			expected: true,
+		},
+		{
+			name: "accept longer pre-release with same prefix",
+			added: "0.0.1-1.2",
+			compare: "0.0.1-1",
+			expected: true,
+		},
+		{
+			name: "accept numerically larger pre-release",
+			added: "0.0.1-11",
+			compare: "0.0.1-2",
+			expected: true,
+		},
+		{
+			name: "accept lexicographically larger pre-release",
+			added: "0.0.1-beta.1",
+			compare: "0.0.1-alpha.1",
+			expected: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			added, err := semver.Make(tt.added)
+			require.NoError(t, err)
+			compare, err := semver.Make(tt.compare)
+			require.NoError(t, err)
+			actual := isSkipPatchCandidate(added, compare)
+			assert.Equal(t, tt.expected, actual)
+
+			if !tt.commutative {
```

Comment:
Now that I see it (and recognize that there are cases that don't have this property), I'm not sure that adding a second assertion to the test is better than explicitly writing these cases. I won't ask you to change it again if you disagree, though.

Reply:
We can make this assertion whenever we have a valid skippatch pair. `f(a, c) -> !f(c, a)` holds true, so this should cover all the cases we need